quiz2/app/controllers/ideas_controller.rb

```diff
@@ -0,0 +1,63 @@
+class IdeasController < ApplicationController
+    before_action :find_idea, except: [:index, :new, :create]
+    before_action :authenticate_user!, except: [:index, :show]
```

Comment:
`authenticate_user!` and `authorize_user!` need to be defined in application controller so that likes and reviews controllers get an access to auth as well.